src/components/timeline/ActiveTabResourceTrack.js

```diff
@@ -58,22 +58,26 @@ class ActiveTabResourceTrackComponent extends PureComponent<Props, State> {
     };
   }
 
-  _onMouseUp = (event: MouseEvent) => {
+  _onMouseUp = fromWhere => (event: MouseEvent) => {
     const { isSelected } = this.props;
     const { isOpen } = this.state;
 
     if (event.button === 0) {
       // Don't allow clicks on the threads list to steal focus from the tree view.
       event.preventDefault();
-      if (isSelected || !isOpen) {
-        // We have two different states for selected tracks and open tracks because
-        // non selected tracks also can stay opened. We can only close the track if
-        // the track is already selected.
-        this.setState(prevState => {
-          return { isOpen: !prevState.isOpen };
-        });
+      if (fromWhere === 'timelineTrackResourceLabel') {
+        if (isSelected || !isOpen) {
+          // We have two different states for selected tracks and open tracks because
+          // non selected tracks also can stay opened. We can only close the track if
+          // the track is already selected.
+          this.setState(prevState => {
+            return { isOpen: !prevState.isOpen };
+          });
+        }
+        this.props.selectActiveTabTrack(this._getTrackReference());
+      } else {
+        this.props.selectActiveTabTrack(this._getTrackReference());
       }
```

Comment:
We don't need that else branch, we can move the upper `this.props.selectActiveTabTrack(this._getTrackReference());` outside of the if directly since both `if` and `else` branches share the same code.